File: app/src/main/java/com/semisonfire/cloudgallery/ui/main/disk/adapter/items/UploadItem.java
package com.semisonfire.cloudgallery.ui.main.disk.adapter.items;

import com.semisonfire.cloudgallery.data.model.Photo;

import java.util.ArrayList;
import java.util.List;

public class UploadItem extends DiskItem {

    private List<Photo> uploadPhotos;
    private int size;
    private int uploadCount;
    private int visibility;
    private String state;

    public UploadItem() {
        uploadPhotos = new ArrayList<>();
        uploadCount = 1;
    }

    public List<Photo> getUploadPhotos() {
        return uploadPhotos;
    }

    public void addUploadPhotos(List<Photo> photos) {
        uploadPhotos.addAll(photos);
        size = uploadPhotos.size();
    }

    public void resetUploadCount() {
        this.uploadCount = 1;
    }

    public void incrementUpload() {
        ++uploadCount;
    }

    @Override
    public int getType() {
        return TYPE_UPLOAD;
    }

    public int getVisibility() {
        return visibility;
    }

    public void setVisibility(int visibility) {
        this.visibility = visibility;
    }

    public int getSize() {
        return size;
    }

    public void setSize(int size) {
        this.size = size;
    }

    public int getUploadCount() {
        return uploadCount;
    }

    public String getState() {
        return state;
    }

    public void setState(String state) {
        this.state = state;
    }

    @Override
    public boolean equals(Object obj) {

        if (obj instanceof UploadItem) {
            return getType() == ((UploadItem) obj).getType();
        }

        return super.equals(obj);
    }
}

File: app/src/main/java/com/semisonfire/cloudgallery/ui/disk/adapter/items/UploadItem.kt
package com.semisonfire.cloudgallery.ui.disk.adapter.items

import com.semisonfire.cloudgallery.data.model.Photo

class UploadItem : DiskItem() {
    private val uploadPhotos: MutableList<Photo>

    var size = 0
    var uploadCount: Int
        private set
    var visibility = 0
    var state: String? = null

    fun getUploadPhotos(): MutableList<Photo> {
        return uploadPhotos
    }

    fun addUploadPhotos(photos: List<Photo>) {
        uploadPhotos.addAll(photos)
        size = uploadPhotos.size
    }

    fun resetUploadCount() {
        uploadCount = 1
    }

    fun incrementUpload() {
        ++uploadCount
    }

    override val type: Int
        get() = TYPE_UPLOAD

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is UploadItem) return false

        if (type != other.type) return false

        return true
    }

    override fun hashCode(): Int {
        return type
    }

    init {
        uploadPhotos = ArrayList()
        uploadCount = 1
    }
}